vm/src/stdlib/scproxy.rs

```diff
@@ -0,0 +1,140 @@
+pub(crate) use _scproxy::make_module;
+
+#[pymodule]
+mod _scproxy {
+    // straight-forward port of Modules/_scproxy.c
+
+    use crate::builtins::{PyDictRef, PyStr};
+    use crate::pyobject::{IntoPyObject, ItemProtocol, PyResult};
+    use crate::VirtualMachine;
+    use std::ops::Deref;
+    use system_configuration::core_foundation::{
+        array::CFArray,
+        base::TCFType,
+        dictionary::CFDictionary,
+        number::CFNumber,
+        string::{CFString, CFStringRef},
+    };
+    use system_configuration::sys::{
+        dynamic_store_copy_specific::SCDynamicStoreCopyProxies, schema_definitions::*,
+    };
+
+    #[pyfunction]
+    fn _get_proxy_settings(vm: &VirtualMachine) -> PyResult<Option<PyDictRef>> {
+        // Py_BEGIN_ALLOW_THREADS
+        let proxy_dict_raw = unsafe { SCDynamicStoreCopyProxies(std::ptr::null()) };
+        // Py_END_ALLOW_THREADS
+        if proxy_dict_raw.is_null() {
+            return Ok(None);
+        }
+        let proxy_dict =
+            unsafe { CFDictionary::<_, CFNumber>::wrap_under_get_rule(proxy_dict_raw) };
+
+        let result = vm.ctx.new_dict();
+        let v = 0
+            != if let Some(a_num) =
+                unsafe { proxy_dict.find(kSCPropNetProxiesExcludeSimpleHostnames) }
```

Comment:
I don't think this `unsafe` block is necessary? It looks like `CFDictionary::find` is safe to me from the docs: https://docs.rs/core-foundation/0.7.0/core_foundation/dictionary/struct.CFDictionary.html#method.find

Reply:
Ah, that makes sense.